### ISSUE TITLE
Cached formula string type ignored

### ISSUE THREAD

**Issue by Jaecen** [ExcelDataReader/ExcelDataReader#53]
SpreadsheetML provides a "str" cell type on formula cells that indicates that the cached formula result should be read as a string. ExcelDataReader does not have support for this cell type and instead handles it as an untyped cell, converting the value to a double if .NET will parse it. Instead, it should be handled similar to an inline string.

Additionally, when ExcelOpenXmlReader reads a cell value it first [converts it to to a double if possible](../tree/2db357e6db7280a122fa27730baa17151095b2d6/Excel/ExcelOpenXmlReader.cs#L258-259). Then, if it determines that the cell was an inline string, it [converts the double back to a string](../tree/2db357e6db7280a122fa27730baa17151095b2d6/Excel/ExcelOpenXmlReader.cs#L265-268). This can cause loss of information, such as leading zeros and trailing commas and periods. Instead, inline string types should never be converted to a double. I believe this is similar to issues #47 and #31.

**Comment by killergege** [comment]
I have this exact same issue. 
I have a formula that gets a code that starts with zeros, and those are removed when read by ExcelDataReader.

I see someone seem to have fixed it. It would be great to include that fix in the released version !

**Comment by wizardgsz** [comment]
Sure, can you kindly double check if the proposed solution work fine for your Excel data file?
With regards and thanks!

**Comment by killergege** [comment]
Ok, I applied (manually - I don't have a Git client here) the changes from the commit above on the right files (but ignored the changes on test files) to the current most recent version on GitHub and tried it with the TestApp included.

I confirm that these changes fixes the specific issue that removes the leading zeros when the cell content is obtained from a formula.
I only tested this case, but it works as expected for me.
